python/src/dataset.rs

```diff
@@ -152,6 +152,34 @@ impl Dataset {
                 .map_err(|err| PyIOError::new_err(err.to_string()))?)
         })
     }
+
+    fn versions(self_: PyRef<'_, Self>) -> PyResult<Vec<PyObject>> {
+        let versions = self_
+            .list_versions()
+            .map_err(|err| PyValueError::new_err(err.to_string()))?;
+        Python::with_gil(|py| {
+            let pyvers: Vec<PyObject> = versions
+                .iter()
+                .map(|v| {
+                    let dict = PyDict::new(py);
+                    dict.set_item("version", v.version).unwrap();
```

Comment:
what does `dict.set_item()` return? `Result` or `Option`? Curious why it needs to unwrap().

Reply:
Result